network-area-diagram/src/main/java/com/powsybl/nad/svg/iidm/AbstractVoltageStyleProvider.java

```diff
@@ -117,4 +125,73 @@ protected Optional<String> getBaseVoltageStyle(BranchEdge edge, BranchEdge.Side
     }
 
     protected abstract Optional<String> getBaseVoltageStyle(Terminal terminal);
+
+    @Override
+    public List<String> getHighlightNodeStyleClasses(Node node) {
+        String subnetworkId = subnetworkEquipmentMap.get(node.getEquipmentId());
+        return List.of(subnetworksHighlightMap.get(subnetworkId));
+    }
+
+    @Override
+    public List<String> getHighlightSideEdgeStyleClasses(BranchEdge edge, BranchEdge.Side side) {
+        String subnetworkId = getSubnetworkId(edge, side);
+        return List.of(subnetworksHighlightMap.get(subnetworkId));
+    }
+
+    @Override
+    public List<String> getHighlightThreeWtEdgStyleClasses(ThreeWtEdge edge) {
+        String subnetworkId = getSubnetworkId(edge.getEquipmentId(), edge.getSide());
+        return List.of(subnetworksHighlightMap.get(subnetworkId));
+    }
+
+    private String getSubnetworkId(BranchEdge edge, Side side) {
+        String subnetworId = null;
+        switch (edge.getType()) {
+            case BranchEdge.LINE_EDGE:
+                TwoSides lineSide = side.equals(Side.ONE) ? TwoSides.ONE : TwoSides.TWO;
+                subnetworId = network.getLine(edge.getEquipmentId()).getTerminal(lineSide).getVoltageLevel().getParentNetwork().getId();
+                break;
+            case BranchEdge.TWO_WT_EDGE, BranchEdge.PST_EDGE:
+                TwoSides twSide = side.equals(Side.ONE) ? TwoSides.ONE : TwoSides.TWO;
+                subnetworId = network.getTwoWindingsTransformer(edge.getEquipmentId()).getTerminal(twSide).getVoltageLevel().getParentNetwork().getId();
+                break;
+            case BranchEdge.DANGLING_LINE_EDGE:
+                subnetworId = network.getDanglingLine(edge.getEquipmentId()).getTerminal().getVoltageLevel().getParentNetwork().getId();
+                break;
+            case BranchEdge.TIE_LINE_EDGE:
+                subnetworId = network.getTieLine(edge.getEquipmentId()).getTerminal(side.equals(Side.ONE) ? TwoSides.ONE : TwoSides.TWO).getVoltageLevel().getParentNetwork().getId();
+                break;
+            case BranchEdge.HVDC_LINE_EDGE:
+                subnetworId = network.getHvdcLine(edge.getEquipmentId()).getConverterStation(side.equals(Side.ONE) ? TwoSides.ONE : TwoSides.TWO).getTerminal().getVoltageLevel().getParentNetwork().getId();
+                break;
+            default:
+                break;
+        }
+        return subnetworId;
+    }
+
+    private String getSubnetworkId(String id, ThreeWtEdge.Side side) {
+        return network.getThreeWindingsTransformer(id).getLeg(ThreeSides.valueOf(side.name())).getTerminal().getVoltageLevel().getParentNetwork().getId();
+    }
+
+    private void buildSubnetworkMaps() {
+        List<VoltageLevel> voltageLevels = getVoltageLevels();
```

Comment:
if you only build the subnetwork map, just loop over the `network.getSubnetworks()`